include/boost/url/impl/url_base.hpp

```diff
@@ -47,9 +47,21 @@ class url_base::segments_type::iterator
         bool end) noexcept;
 
 public:
+    using iterator_category =
+        std::bidirectional_iterator_tag;
+
     using value_type =
         segments_type::value_type;
 
+    using pointer =
+        value_type *;
```

Comment:
Since lines 50 and 51 are already as wide as they are there's no need to put `value_type*` on a separate line.

Should this be `value_type const*`?

Reply:
I'll gladly change it, but you'll have to continue to point these one by one, this makes no sense to me.

Reply:
For the formatting I don't care enough to want it to be changed, but what about `value const*`?

Reply:
Oh I missed that, sorry. Yes, also `reference` should be `value_type const &` right?

Reply:
yeah but if you could put `const&` together instead of `const &` (and ditto for `const*`) it would be great

Reply:
Sure. Should I replace it elsewhere too? I may have put spaces before `*` and `&` in other places.

Reply:
yes please